pandas/_libs/include/pandas/vendored/klib/khash_python.h

```diff
@@ -2,10 +2,14 @@
 #include <Python.h>
 
 
-// use numpy's definitions for complex
-#include <numpy/arrayobject.h>
-typedef npy_complex64 khcomplex64_t;
-typedef npy_complex128 khcomplex128_t;
+typedef struct {
+    float real;
+    float imag;
+    } khcomplex64_t;
+typedef struct {
+    double real;
+    double imag;
```

Comment:
will this change anything on platforms where float/double aren't 32/64 bits?  (which i guess we have zero testing for so not a huge deal)

Reply:
Not sure. It appears in https://github.com/numpy/numpy/pull/24085 in numpy/core/include/numpy/npy_math.h numpy also defines real/imag components as doubles

Reply:
I think the misnomer is to call these complex64 and complex128. The C99 standard has float complex and double complex types, but nothing that is a fixed width

See also https://stackoverflow.com/questions/2524737/fixed-size-floating-point-types#2524933

Reply:
Yeah, I guess technically float32/float64 are sized aliases under the assumption they refer to float/double, but in practice the assumption seems so strong that I doubt it matters.

I am very puzzled about the change though, this is exactly what it was after all, and I don't think C cares about type aliases (even if there was interaction anywhere).  I am wondering if the `include <numpy/arrayobject.h>` somehow is involved, but I don't see _why_ it should be either?

I assume the value stored is correct here and just the printing somehow thinks it is printing a NaN?